src/components/numeral-date/numeral-date.component.tsx

```diff
@@ -17,23 +17,126 @@ import { TooltipProvider } from "../../__internal__/tooltip-provider";
+  adaptiveLabelBreakpoint?: number;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-component"?: string;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-element"?: string;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-role"?: string;
+  /** If true, the component will be disabled */
+  disabled?: boolean;
+  /** If true, the component will be read-only */
+  readOnly?: boolean;
+  /* Array of strings to define custom input layout.
+  Allowed formats:
+  ['dd', 'mm', 'yyyy'],
+  ['mm', 'dd', 'yyyy'],
+  ['dd', 'mm'],
+  ['mm', 'dd'],
+  ['mm', 'yyyy'] */
+  dateFormat?: ValidDateFormat;
+  /** Default value for use in uncontrolled mode  */
+  defaultValue?: NumeralDateObject;
+  /**  Value for use in controlled mode  */
+  value?: NumeralDateObject;
+  /** When true, enables the internal errors to be displayed */
+  enableInternalError?: boolean;
+  /** When true, enables the internal warnings to be displayed */
+  enableInternalWarning?: boolean;
+  /** Help content to be displayed under an input */
+  fieldHelp?: React.ReactNode;
+  /** `id` for events */
+  id?: string;
+  /** `name` for events */
+  name?: string;
+  /** Label */
+  label?: string;
+  /** Label alignment. Works only when labelInline is true */
+  labelAlign?: "left" | "right";
+  /** Text applied to label help tooltip */
+  labelHelp?: React.ReactNode;
+  /** When true, label is placed in line with an input */
+  labelInline?: boolean;
+  /** Label width */
+  labelWidth?: number;
+  /** Spacing between label and a field for inline label, given number will be multiplied by base spacing unit (8) */
+  labelSpacing?: 1 | 2;
+  /** Blur event handler */
+  onBlur?: (ev: NumeralDateEvent) => void;
+  /** Change event handler */
+  onChange?: (ev: NumeralDateEvent) => void;
+  /** Flag to configure component as mandatory */
+  required?: boolean;
+  /** Size of an input */
+  size?: "small" | "medium" | "large";
+  /** When true, validation icons will be placed on labels instead of being placed on the inputs */
+  validationOnLabel?: boolean;
+  /** Overrides the default tooltip position */
+  tooltipPosition?: "top" | "bottom" | "left" | "right";
+  /** Aria label for rendered help component */
+  helpAriaLabel?: string;
+}
+
+const ALLOWED_DATE_FORMATS: ValidDateFormat[] = [
   ["dd", "mm", "yyyy"],
   ["mm", "dd", "yyyy"],
   ["dd", "mm"],
   ["mm", "dd"],
   ["mm", "yyyy"],
 ];
 
-const isDayValid = (day) => (day ? +day > 0 && +day < 32 : true);
+const incorrectDateFormatMessage =
+  "Forbidden prop dateFormat supplied to NumeralDate. " +
+  "Only one of these date formats is allowed: " +
+  "['dd', 'mm', 'yyyy'], " +
+  "['mm', 'dd', 'yyyy'], " +
+  "['dd', 'mm'], " +
+  "['mm', 'dd'], " +
+  "['mm', 'yyyy']";
+
+const isDayValid = (day: string) => (day ? +day > 0 && +day < 32 : true);
```

Comment:
Not a problem for this PR but if we're baking in validations here I wonder why we aren't using `Date` here to check the days are valid for a given month and year 🤷‍♂️ 

```
const isDayValid = (year, month, day) => {
  const computedDay = new Date(`${+year}/${+month}/${+day}`).getDate();
  return computedDay === day;
}
```

Reply:
Yeah was just a comment rather than something I expected you to change here, I'm not convinced it's a breaking change as to me that is a bug in our internal validations but again not a problem for this PR